src/services/ansibleLint.ts

```diff
@@ -131,7 +136,15 @@ export class AnsibleLint {
               workingDirectory
             );
           } else {
+            // Notifying the user about the failed command and falling back to ansible syntax-check in this scenario
+            this.connection.window.showWarningMessage(
+              'Falling back to ansible syntax-check.'
+            );
             this.connection.window.showErrorMessage(execError.message);
+            console.debug(
+              'Ansible-lint command execution failed. Falling back to ansible syntax-check'
+            );
```

Comment:
These are some useful logs to see at what point fallback to ansible syntax-check happens.

Reply:
Still, the two `connection.window` messages could be combined.

Reply:
This is kept this way purposely.
Both the notifications have separate purposes and thus I believe combining them is not a good idea (they are intended to appear at the same time, on top of one another).

Reply:
Alright, in that case there is no reason not to bubble up that `'Falling back to ansible syntax-check.'` to where it belongs - in the `validationProvider`, which actually has logic to fall back to syntax-check. We should leave this provider without such logs/knowledge to achieve lower coupling of these two providers (a generally good software development practice).

Reply:
Since we don't have direct access to the `connection` attribute at the provider level @priyamsahoo and I discussed two approaches to do it
1) Change the `connection` attribute from private to public in `ValidationManager` class so that it can be accessed in `doValidate` function of `validationProvider`
2) Pass around `connection` object from `anisbleLanguageService` to `doValidate` function call.

If there is any other better please let us know. If we end up going this route I will prefer option 2. WDYT?


Both the approaches are a deviation from the way the existing providers are structured. I
> We should leave this provider without such logs/knowledge to achieve lower coupling of these two providers (a generally good software development practice).

Agree with this comment about keeping services independent. IMO showing an informational message of falling back to syntax check in case ansible-lint fails is an useful information to users and I believe instances of this occurring is rare. In any case, since this is only a log message and not an invocation of an object from one service to another I am fine with the way it is currently.

Reply:
I would also recommend option 2. I too think that the message is useful, but it's hardly the place where the decision/logic expressed in that message happens. Maybe only a log message, but already a foothold and invitation for more caller-dependent code in the future ;) I know it is not easy to implement well encapsulated code, but it's as with this saying: "bug you now or bug you later".